packages/react-admin/src/reducer/admin/references/possibleValues.js

```diff
@@ -16,22 +16,16 @@ export default (previousState = initialState, { type, payload, meta }) => {
 
 export const getPossibleReferences = (
     state,
-    referenceSource,
-    reference,
+    possibleValues,
     selectedIds = []
 ) => {
-    const possibleValues = state.admin.references.possibleValues[
-        referenceSource
-    ]
-        ? Array.from(state.admin.references.possibleValues[referenceSource])
-        : [];
+    possibleValues = possibleValues ? Array.from(possibleValues) : [];
     selectedIds.forEach(
         id =>
             possibleValues.some(value => value == id) ||
             possibleValues.unshift(id)
     );
-
     return possibleValues
-        .map(id => state.admin.resources[reference].data[id])
+        .map(id => state.data[id])
```

Comment:
what's `state.data`? it doesn't exist.

Reply:
it's a partial state received from the reselect selector created in `ReferenceInput`

Reply:
OK, then don't use the name `state`, it's misleading.